### PR TITLE
Feature/tech 307 hide k8s details

### PR DESCRIPTION
**Pull request by Jorg88**

### CODE REVIEW

File: src/mpyl/steps/deploy/k8s/rancher.py [hunk 1]
```diff
@@ -2,32 +2,40 @@
 
 from dataclasses import dataclass
 
+from ....steps import Input
 from ....project import Target
 
 
-@dataclass(frozen=True)
```
Comment (koeves) [inline src/mpyl/steps/deploy/k8s/rancher.py:8]: Why is it not frozen now?

File: src/mpyl/steps/deploy/k8s/rancher.py [hunk 2]
```diff
@@ -2,32 +2,40 @@
...
+@dataclass()
 class ClusterConfig:
     project_id: str
     cluster_id: str
     cluster_env: str
     context: str
 
+    def __init__(self, config: dict):
```
Comment (koeves) [inline src/mpyl/steps/deploy/k8s/rancher.py:16]: No need to create an __init__ in a dataclass

**Review by koeves — COMMENTED** [review]
Looks good just small comments

File: src/mpyl/steps/deploy/k8s/rancher.py (hunk 1 above)
Comment (Jorg88) [reply src/mpyl/steps/deploy/k8s/rancher.py:8]: See thread in other comment

File: src/mpyl/steps/deploy/k8s/rancher.py (hunk 2 above)
Comment (Jorg88) [reply src/mpyl/steps/deploy/k8s/rancher.py:16]: Yeah I wasn't sure about this part, how would you apply the logic that's currently in the `init` then? This is more clean than having to pass `cluster_configs['test'].projectId, cluster_configs['test'].clusterId, etc` for all 3 environments.

Comment (koeves) [reply src/mpyl/steps/deploy/k8s/rancher.py:16]: hmm yeah fair enough.. I think in other places we do this as well, and call the method `from_config()` or so

Comment (koeves) [reply src/mpyl/steps/deploy/k8s/rancher.py:16]: And it can be frozen as well

**Review by koeves — APPROVED** [review]

**Pull request merged**